src/app/qgisapp.cpp

```diff
@@ -1064,6 +1064,8 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, bool skipBadLayers
     settings.setValue( QStringLiteral( "qgis/main_canvas_preview_jobs" ), true );
   }
   mMapCanvas->setPreviewJobsEnabled( settings.value( QStringLiteral( "qgis/main_canvas_preview_jobs" ), true ).toBool() );
+  // record profiling time on the main canvas only
+  mMapCanvas->mapSettings().setFlag( Qgis::MapSettingsFlag::RecordProfile );
```

Comment:
Out of curiosity, is there a cost associated with this profiling, and if so do we want to allow for it to be toggled on / off?

Reply:
It should be negligible -- at most each layer render will trigger a handful of signals and some cheap QElapsedTimer calculations. 

The main reason for limiting it to the map canvas only is that I don't want results shown for other canvas renders (eg the CRS selector extent preview widget) .